src/catalog/table_schema.cpp

```diff
@@ -0,0 +1,198 @@
+#include "catalog/table_schema.h"
+
+#include "common/exception.h"
+#include "common/ser_deser.h"
+#include "common/string_utils.h"
+
+using namespace kuzu::common;
+
+namespace kuzu {
+namespace catalog {
+
+RelMultiplicity getRelMultiplicityFromString(const std::string& relMultiplicityString) {
+    if ("ONE_ONE" == relMultiplicityString) {
+        return RelMultiplicity::ONE_ONE;
+    } else if ("MANY_ONE" == relMultiplicityString) {
+        return RelMultiplicity::MANY_ONE;
+    } else if ("ONE_MANY" == relMultiplicityString) {
+        return RelMultiplicity::ONE_MANY;
+    } else if ("MANY_MANY" == relMultiplicityString) {
+        return RelMultiplicity::MANY_MANY;
+    }
+    throw CatalogException("Invalid relMultiplicity string '" + relMultiplicityString + "'.");
+}
+
+std::string getRelMultiplicityAsString(RelMultiplicity relMultiplicity) {
+    switch (relMultiplicity) {
+    case RelMultiplicity::MANY_MANY: {
+        return "MANY_MANY";
+    }
+    case RelMultiplicity::MANY_ONE: {
+        return "MANY_ONE";
+    }
+    case RelMultiplicity::ONE_ONE: {
+        return "ONE_ONE";
+    }
+    case RelMultiplicity::ONE_MANY: {
+        return "ONE_MANY";
+    }
+    default:
+        throw CatalogException("Cannot convert rel multiplicity to std::string.");
+    }
+}
+
+void Property::serialize(FileInfo* fileInfo, uint64_t& offset) const {
+    SerDeser::serializeValue<std::string>(name, fileInfo, offset);
```

Comment:
Actually, you don't need to write the specialization type here. C++ will auto infer the correct function based on the type of name